src/hayhooks/server/utils/mcp_utils.py

```diff
@@ -48,9 +41,15 @@ class CoreTools(str, Enum):
 
 if TYPE_CHECKING:
     from mcp.types import TextContent, ImageContent, EmbeddedResource, Tool
+    from mcp.server import Server
```

Comment:
just a question. The `LazyImport` block below is not sufficient to ensure type checking?
what if we remove this `if TYPE_CHECKING:` section?

Reply:
Hmm not sure we can safely remove, since `mcp` is an optional dep and below on that file we're using unquoted type names. Theoretically, we could remove it and quote unquoted type names.

BTW, I would first do [this](https://github.com/deepset-ai/hayhooks/issues/95) and then re-evaluate if we can safely remove it. ATM there's no reliable type checking and linting (we should add this to CI). WDYT?